neuro cp -r doesn't work with non directories

#### Use-case:
I expect to use something like this for code uploading
```neuro cp -r * $(CODE_FOLDER)```

But get 
```neuro cp -r * storage:src/text_summarization/dev
Copy file:///.../Programs/Neuromation/text_summarization/Dockerfile => storage://rauf-kurbanov/src/text_summarization/dev/Dockerfile
ERROR: [Errno 20] Not a directory: '/Users/raufkurbanov/Programs/Neuromation/text_summarization/Dockerfile'
make: *** [update_code] Error 72
```

#### Expected behavior:

It's expected for unix user to use -r flag even with non-directories
```
➜  /tmp touch a.txt
➜  /tmp mkdir b
➜  /tmp cp -r a.txt b
```

--- a/CHANGELOG.D/1053.feature
+++ b/CHANGELOG.D/1053.feature
@@ -1,0 +1,1 @@
+`neuro cp -r` now works with non-directories.

--- a/neuromation/cli/storage.py
+++ b/neuromation/cli/storage.py
@@ -300,7 +300,7 @@ async def cp(
         progress_obj.begin(src, dst)
 
         if src.scheme == "file" and dst.scheme == "storage":
-            if recursive:
+            if recursive and await root.client.storage._is_dir(src):
                 await root.client.storage.upload_dir(
                     src, dst, update=update, progress=progress_obj
                 )
@@ -309,7 +309,7 @@ async def cp(
                     src, dst, update=update, progress=progress_obj
                 )
         elif src.scheme == "storage" and dst.scheme == "file":
-            if recursive:
+            if recursive and await root.client.storage._is_dir(src):
                 await root.client.storage.download_dir(
                     src, dst, update=update, progress=progress_obj
                 )

--- a/tests/e2e/test_e2e_storage.py
+++ b/tests/e2e/test_e2e_storage.py
@@ -359,6 +359,23 @@ def test_e2e_copy_recursive_to_platform(
 
 
 @pytest.mark.e2e
+def test_e2e_copy_recursive_file(helper: Helper, tmp_path: Path) -> None:
+    srcfile = tmp_path / "testfile"
+    dstfile = tmp_path / "copyfile"
+    srcfile.write_bytes(b"abc")
+
+    captured = helper.run_cli(["storage", "cp", "-r", str(srcfile), helper.tmpstorage])
+    assert not captured.out
+
+    captured = helper.run_cli(
+        ["storage", "cp", "-r", helper.tmpstorage + "testfile", str(dstfile)]
+    )
+    assert not captured.out
+
+    assert dstfile.read_bytes() == b"abc"
+
+
+@pytest.mark.e2e
 def test_e2e_rename(helper: Helper) -> None:
     helper.check_create_dir_on_storage("folder")
     helper.run_cli(

